src/messages.rs

```diff
@@ -4,13 +4,23 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-use crate::util::pubkey::PublicKey;
+use crate::{commons::Topics, util::pubkey::PublicKey};
+use bytes::{Buf, BufMut, Bytes, BytesMut};
+use dusk_bytes::DeserializableSlice;
+
 pub enum Status {
     Past,
     Present,
     Future,
 }
+///
+pub trait Serializable {
+    /// Serialize struct to Vec<u8>.
+    fn to_bytes(&self) -> Vec<u8>;
 
+    /// Deserialize struct from buf by consuming N bytes.
+    fn from_bytes(buf: &mut Bytes) -> Self;
```

Comment:
The from_bytes should be aware of errors, otherwise a malicious message can be crafted to let the consensus panic.
Furthermore I would prefer to not use Bytes but a standard Read as it's done in https://github.com/dusk-network/kadcast/blob/c46d5e78ca0587226454bf753a53ca295ddf5930/src/encoding/payload/broadcast.rs#L25

Reply:
WRT malicious/malformed message, I'm aware of that.  It's a valid point, however this can be addressed when fixing https://github.com/dusk-network/consensus/issues/22. This PR ultimate goal is to allow us to test consensus in the context of kadcast cluster and also build a pilot node implementation.